homeassistant/components/climate/generic_thermostat.py

```diff
@@ -121,14 +129,20 @@ def __init__(self, hass, name, heater_entity_id, sensor_entity_id,
     @asyncio.coroutine
     def async_added_to_hass(self):
         """Run when entity about to be added."""
-        # If we have an old state and no target temp, restore
-        if self._target_temp is None:
-            old_state = yield from async_get_last_state(self.hass,
-                                                        self.entity_id)
-            if old_state is not None:
+        # Check If we have an old state
+        old_state = yield from async_get_last_state(self.hass,
+                                                    self.entity_id)
+        if old_state is not None:
+            # If we have no initial temperature, restore
+            if self._target_temp is None:
                 self._target_temp = float(
                     old_state.attributes[ATTR_TEMPERATURE])
 
+            # If we have no initial operation mode, restore
+            if self._initial_operation_mode is None:
+                if old_state.attributes[ATTR_OPERATION_MODE] == STATE_OFF:
+                    self.set_operation_mode(STATE_OFF)
```

Comment:
We can not do this. The restore of things only set internal variable but do not make any actions. We don't know in which initial process that will be executed, maybe it exists not on this time and we run into a wired case. You can only modify the `self._enabled` with `async_added_to_hass`.

Reply:
If I modify only `self._enabled` it work but there are an issue:

If for some reason the heater switch is restored to On at startup and generic_thermostat to Off, the switch is turned off only after the keep alive time. If the user don't have a keep alive time in configuration the switch is never turned off.

We need something that trigger `set_operation_mode` at least one time when generec_thermostat start.

I made many tests and never got errors calling  `self.set_operation_mode(STATE_OFF)` from  `async_added_to_hass `. 

Any suggestion?